enhancements/sdk-kubernetes-1.21.md

```diff
@@ -0,0 +1,219 @@
+
+_GitHub: [operator-framework/api][gh-api]_
+
+##### Graduation Criteria
+
+- Bumping the controller-runtime version in go.mod scaffolds to the controller-runtime version that supports Kubernetes 1.21.
+- Updating tests to use Kubernetes 1.21 where possible and create follow-on tasks for tests that cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade operator-framework/java-operator-plugins to support 1.21
+
+_GitHub: [operator-framework/java-operator-plugins][gh-java]_
+
+##### Graduation Criteria
+
+- Bumping the apimachinery and kubebuilder version in `go.mod` to a version that supports Kubernetes 1.21.
+- Updating e2e tests to use Kubernetes 1.21 where possible and create follow-on tasks for tests that cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
+[gh-controller-runtime]: https://github.com/kubernetes-sigs/controller-runtime
+[gh-controller-tools]: https://github.com/kubernetes-sigs/controller-tools
+[gh-kubebuilder]: https://github.com/kubernetes-sigs/kubebuilder
+[gh-operator-sdk]: https://github.com/operator-framework/operator-sdk
+[gh-operator-lib]: https://github.com/operator-framework/operator-lib
+[gh-api]: https://github.com/operator-framework/api
+[gh-java]: https://github.com/operator-framework/java-operator-plugins
+
+### Risks and Mitigations
+
+- As always there is risk to the actual development and delivery timing of this feature, due to the many upstream changes that are required to be completed before Operator SDK can support Kubernetes
+1.21.
+
+| Repositories where k8s is to bumped to 1.21        | Estimated timeline    |
+| -------------------------------------------------- | --------------------- |
+| Controller-runtime                                 | Done                  |
+| Controller-tools                                   | Done                  |
+| Kubebuilder binary, v3 plugin and kb-declarative pattern                   | ~ July 2              |
+| Operator-framework/api                             | ~ July 2              |
+| Operator-framework/operator-lib                    | ~ July 2              |
+| Operator SDK                                       | ~ July 22             |
+| Operator-framework/java-operator-plugin            | ~ July 22             |
+
+Given the estimated timeline above, it can be expected that by the third week July, we will have an upstream Operator SDK release which will support Kubernetes 1.21.
+
+- In particular, as the Operator SDK release relates to downstream integrations (e.g. OpenShift), the current OpenShift 4.9 release timelines have feature freeze around September 3. This translates that we would only have around 10 days to cut a [downstream SDK](https://github.com/openshift/ocp-release-operator-sdk) release which supports k8s 1.21.
```

Comment:
This is something that Red Hat folks should consider, but doesn't belong in an upstream EP.
```suggestion
```

Reply:
oh..okay. Then maybe it would be better if I remove the downstream timeline from here, and just list out the expected dates when an upstream release with k8s 1.21 can be expected ?

Reply:
Idont think we need to commit to dates in an EP. As RH employees, we should consider downstream releases for our own prioritization, but IMO pure upstream users dont need to worry about it.

Reply:
Removed the specifics about syncing downstream. Apart from that I don't see any other risks, have noted it down.